power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -12,8 +12,11 @@
 #include "dataset_fwd.hpp"
 #include "meta_data.hpp"
 
+#include <boost/range.hpp>
+
 #include <span>
 #include <string_view>
+#include <variant>
```

Comment:
currently not used

Reply:
It is used at `template <typename StructType> using BufferSpanType = std::variant<std::span<StructType>, RangeObject<StructType>>;`

Reply:
Why use a variant?

Reply:
I was trying to combine together `get_columnar_buffer_span` and `get_buffer_span` into one, as much of the code between both was repeated, hence I used `variant` to hold both possible return types. However, now I am having many thoughts about this.

Reply:
I don't think we need `variant` here. Just define two different functions with different return type. In both function you put a `assert` to check if the user is calling the correct function.

The user is supposed to call `is_coloumnar` first and then choose to what function to call.

Reply:
Done.